src/UpdateMetadataInterface.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth;
+
+/**
+ * Describes a Credentials object which supports updating request metadata
+ * (request headers).
+ */
+interface UpdateMetadataInterface
```

Comment:
Should the `IAMCredentials` class implement this too? The auth keys seem to differ in that implementation a bit though, which makes it seem like the constant being on this class might not be ideal.

Reply:
I don't really know how the `IAMCredentials` class is being used, so I left it out for now. It doesn't even extend `FetchAuthTokenInterface` so it makes more sense for me to leave it out.

Reply:
Do we need the interface then? Perhaps I missed it but it looks like anything that extends `CredentialsLoader` will implement the `updateMetadata` method. If the goal is to provide better implementation checking, we can do that by checking if the credentials are an instance of `CredentialsLoader` I believe.

Reply:
After thinking on this a bit, the interface is better, let's keep it.

Reply:
The interface is used by `FetchAuthTokenCache`, which does not extend `CredentialsLoader`. It's only real function is to ensure that the `updateMetadata` function exists on the fetcher, since we can't add that method to `FetchAuthTokenInterface` without breaking backwards compatibility.

It isn't strictly necessary but I think it's best practice to have it. Not only is the interface cleaner, but it ensures that the function signature is compatible.